src/main/java/org/minions/devfund/katerinaanzoleaga/moviedatabase/MovieDatabase.java

```diff
@@ -0,0 +1,280 @@
+        }
+    }
+
+    /**
+     * Gets the Actor with best average rating.
+     * @return
+     */
+    String getBestActor() {
+        double maxAvrRating = 0;
+        double avrRating = 0;
+        int index = -1;
+        Actor theActor;
+        for (Actor actor: this.actorList) {
+            avrRating = actor.getAverageRating();
+            if (avrRating > maxAvrRating) {
+                maxAvrRating = avrRating;
+                index = this.actorList.indexOf(actor);
+            }
+        }
+        theActor = this.actorList.get(index);
+        return theActor.getName();
+    }
+
+    /**
+     * Gets the best movie based on the max rating.
+     * @return
+     */
+    String getBestMovie() {
+        double maxRating = 0;
+        int index = -1;
+        Movie theMovie;
+        for (Movie movie : this.movieList) {
+            if (movie.getRating() > maxRating){
+                maxRating = movie.getRating();
+                index = this.movieList.indexOf(movie);
+            }
+        }
+        theMovie = this.movieList.get(index);
+        return theMovie.getName();
+
+    }
+
+
+    /**
+     * Fill Movies list from file.
+     * @param fileName
+     */
+    void fillMoviesFromFile(String fileName) {
+        try {
+            File moviesFile = new File(fileName);
+            Scanner scnr = new Scanner(moviesFile);
+            String[] array1;
+            while (scnr.hasNext()) {
+                String aLine = scnr.nextLine();
+                array1 = aLine.split(",");
+
+                String[] actors = new String[1];
+                actors[0] = array1[0].trim();
+                for (int j = 1; j < array1.length; j++) {
+                    int i = this.movieExists(array1[j].trim());
+                    if (i != -1) {
+                        int a = this.actorExists(array1[0].trim());
+                        if (a == -1) {
+                            Actor anActor;
+                            anActor = new Actor();
+                            anActor.setName(array1[0].trim());
+                            anActor.addMovie(this.movieList.get(i));
+                            this.movieList.get(i).addActor(anActor);
+                            this.actorList.add(anActor);
+                        }
+                        else {
+                            this.actorList.get(a).addMovie(this.movieList.get(i));
+                        }
+                    }
+                    else
+                        this.addMovie(array1[j].trim(), actors);
+                }
+            }
+            scnr.close();
+        }
+        catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    /**
+     * Fills the movies rationgs from the file.
+     * @param fileName
+     */
+    void fillRatingsFromFile(String fileName) {
+        try {
+            File ratingsFile = new File(fileName);
+            Scanner scnr = new Scanner(ratingsFile);
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use try-with-resources or close this "Scanner" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2095)